Makefile.include

```diff
@@ -10,6 +10,13 @@ RIOTCPU := $(abspath $(RIOTCPU))
 RIOTBOARD ?= $(RIOTBASE)/boards
 RIOTBOARD := $(abspath $(RIOTBOARD))
 
+# Top level git root of the project being built
+RIOTPROJECT ?= $(shell git rev-parse --show-toplevel)
+RIOTPROJECT := $(abspath $(RIOTPROJECT))
```

Comment:
Regarding building without a git root: Can we defer that use case until an actual user with a different SCM (or no SCM) is asking for it?